app/src/main/java/pl/szczodrzynski/edziennik/ui/modules/timetable/TimetableDayFragment.kt

```diff
@@ -276,6 +287,27 @@ class TimetableDayFragment : LazyFragment(), CoroutineScope {
             lb.detailsFirst.text = listOfNotEmpty(timeRange, classroomInfo).concat(bullet)
             lb.detailsSecond.text = listOfNotEmpty(teacherInfo, teamInfo).concat(bullet)
 
+            val typeIcon = when (attendance?.baseType) {
+                Attendance.TYPE_PRESENT, Attendance.TYPE_PRESENT_CUSTOM -> CommunityMaterial.Icon.cmd_check
+                Attendance.TYPE_ABSENT -> CommunityMaterial.Icon.cmd_close
+                Attendance.TYPE_ABSENT_EXCUSED -> CommunityMaterial.Icon3.cmd_progress_close
+                Attendance.TYPE_RELEASED -> CommunityMaterial.Icon.cmd_account_arrow_right_outline
+                Attendance.TYPE_BELATED -> CommunityMaterial.Icon.cmd_clock_alert_outline
+                Attendance.TYPE_BELATED_EXCUSED -> CommunityMaterial.Icon.cmd_clock_check_outline
+                Attendance.TYPE_DAY_FREE -> SzkolnyFont.Icon.szf_umbrella_beach_outline
+                else -> null
+            }
+
+            if (typeIcon != null) {
+                lb.attendanceIcon.setImageDrawable(
+                    IconicsDrawable(activity, typeIcon).apply {
+                        colorInt = app.attendanceManager.getAttendanceColor(attendance!!)
+                        sizeDp = 24
+                    }
+                )
+                lb.attendanceIcon.isVisible = true
+            }
```

Comment:
```suggestion
            }
            lb.attendanceIcon.isVisible = typeIcon != null
```

Reply:
Yes, but why?

Reply:
To hide the whole element if the icon is not available. Otherwise when a lesson is recycled, the old icon might still show up. An alternative would be to `null` the image drawable if not available, but this is easier.